shared/domains/chelonia/chelonia.js

```diff
@@ -821,7 +823,27 @@ export default (sbp('sbp/selectors/register', {
     const events = await fetch(`${this.config.connectionURL}/eventsAfter/${contractID}/${since}`, { signal: this.abortController.signal })
       .then(handleFetchResult('json'))
     if (Array.isArray(events)) {
-      return events.reverse().map(b64ToStr)
+      // Sanity check
+      if (GIMessage.deserialize(b64ToStr(events[0])).hash() !== since) {
+        throw new Error('hash() !== since')
+      }
+      // Maybe we didn't receive all the requested events because of eventsAfter's limit.
+      if (events.length === MAX_EVENTS_AFTER) {
+        while (true) {
+          const intermediateEventHash = GIMessage.deserialize(b64ToStr(events[events.length - 1])).hash()
+          const nextEvents = await fetch(`${this.config.connectionURL}/eventsAfter/${contractID}/${intermediateEventHash}`)
+            .then(handleFetchResult('json'))
+          // Break if we didn't receive any event we didn't have yet.
+          // Note: nextEvents usually starts with an intermediate event we already have.
+          if (!Array.isArray(nextEvents) || nextEvents.length < 2) break
+          // Avoid duplicating the intermediate event in the result.
+          events.pop()
+          events.push(...nextEvents)
```

Comment:
I know that in some languages, there's a limit to how many arguments you can use in a function call. I don't know if that's true for JavaScript though...

It seems this limit is large and somewhat undefined: https://stackoverflow.com/questions/22747068/is-there-a-max-number-of-arguments-javascript-functions-can-accept

So if we're worried about this, an alternative would be to reassign the variable using `events = events.concat(nextEvents)`